### PR TITLE
[SPARK-21362][SQL][Adding Apache Drill JDBC Dialect]

### PR DESCRIPTION
**Pull request by radford1**
## What changes were proposed in this pull request?

Adding Apache Drill to the JDBC Dialect

## How was this patch tested?
I tested this manually by building and using RStudio to connect to the spark server and running a query against an apache drill cluster

the contribution is my original work and that I license the work to the project under the project’s open source license
Please review http://spark.apache.org/contributing.html before opening a pull request.

### CODE REVIEW

**Comment by radford1** [comment]
I'm very new to this but was just wondering if there is anything else I should do to help with this pull request or does it just take more time?

**Comment by gatorsmile** [comment]
Could you also add the docker-based test suite, like what we did in https://github.com/apache/spark/pull/9893/files?

File: sql/core/src/main/scala/org/apache/spark/sql/jdbc/ApacheDrillDialect.scala [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.spark.sql.jdbc
+
+import java.sql.Types
```
Comment (viirya) [inline sql/core/src/main/scala/org/apache/spark/sql/jdbc/ApacheDrillDialect.scala]: Do we use this import?

File: sql/core/src/main/scala/org/apache/spark/sql/jdbc/ApacheDrillDialect.scala [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+
+import org.apache.spark.sql.types.{BooleanType, DataType, LongType, MetadataBuilder}
```
Comment (viirya) [inline sql/core/src/main/scala/org/apache/spark/sql/jdbc/ApacheDrillDialect.scala]: ditto.

Comment (radford1) [inline sql/core/src/main/scala/org/apache/spark/sql/jdbc/ApacheDrillDialect.scala]: These imports are not used, I will remove them. I'll also look at adding the docker tests suggested by @gatorsmile

**Comment by radford1** [comment]
@viira can you review for me again please

**Comment by viirya** [comment]
@radford1 Have you added the test as suggested by @gatorsmile?

**Comment by radford1** [comment]
@viirya I have not had a chance to build out the docker tests like @gatorsmile suggested. Will that be a requirement in order to merge it?

**Comment by gatorsmile** [comment]
@radford1 Yes. That is needed for merging this PR. Could you check it? Thanks!

**Comment by rxin** [comment]
Unfortunately I think drill is not popular enough to warrant inclusion in here yet. If this is not extensible, we should make it possible to include such mappings outside Spark and then perhaps Drill can just package this (or any 3rd party on github).

**Comment by radford1** [comment]
@gatorsmile Understood I'll work on including them.
@rxin can you help me understand this better? I was trying to add this so spark could call Drill through jdbc. How can we include these mappings outside of spark?

**Comment by rxin** [comment]
We can just put this code in a 3rd party library, can't we? If there is an issue with service/code discovery, we can come up with some sort of registration process similar to the data source API.

**Comment by gatorsmile** [comment]
Could we please close this PR? Thanks!

**Pull request closed without merging**